### PR TITLE
[mlir][arith] Overflow semantics in documentation for muli, subi, and addi

### PR DESCRIPTION
**Pull request by pingshiyu**
Following discussions from this RFC: https://discourse.llvm.org/t/rfc-integer-overflow-semantics

Adding the overflow semantics into the muli, subi and addi arith operations.

### CODE REVIEW

**Comment by pingshiyu** [comment]
@kuhar @regehr @joker-eph @Hardcode84 @math-fehr

File: mlir/include/mlir/Dialect/Arith/IR/ArithOps.td [hunk 1]
```diff
@@ -288,6 +305,17 @@ def Arith_MulIOp : Arith_TotalIntBinaryOp<"muli", [Commutative]> {
...
     Integer multiplication operation.
   }];
+  let description = [{
+    The `muli` operation takes two operands and returns one result, each of
+    these is required to be the same type. This type may be an integer scalar type, 
+    a vector whose element type is integer, or a tensor of integers. It has no 
+    standard attributes.
+
+    The operands are interpreted as unsigned bitvectors. The result is represented by a 
+    bitvector containing the mathematical value of the multiplication modulo 2^n, where `n`
```
Comment (kuhar) [inline mlir/include/mlir/Dialect/Arith/IR/ArithOps.td:315]: nit: would be nice to make this consistent with the notation used in AddUIExtended and MulSIExtended

**Review by Mogball — APPROVED** [review]

**Review by kuhar — APPROVED** [review]
LGTM

**Review by math-fehr — APPROVED** [review]
👍

**Review by Hardcode84 — APPROVED** [review]

**Comment by kuhar** [comment]
@joker-eph do you have any comments?

**Comment by joker-eph** [comment]
No comments, thanks for checking!

**Pull request merged**